src/lib/decodeData.ts

```diff
@@ -312,37 +312,33 @@ export function decodeData(
   data: DecodeDataInput | DecodeDataInput[],
   schema: ERC725JSONSchema[],
 ): DecodeDataOutput | DecodeDataOutput[] {
-  const processDataInput = ({
-    keyName,
-    dynamicKeyParts,
-    value,
-  }: DecodeDataInput) => {
+  const processDataInput = (
+    { keyName, dynamicKeyParts, value }: DecodeDataInput,
+    throwException = true,
+  ) => {
     const isDynamic = isDynamicKeyName(keyName);
 
-    let schemaElement: ERC725JSONSchema;
-    if (isDynamic) {
-      schemaElement = getSchemaElement(schema, keyName, dynamicKeyParts);
-
-      // NOTE: it might be confusing to use as the output will contain other keys as the ones used
-      // for the input
-      return {
-        key: schemaElement.key,
-        name: schemaElement.name,
-        value: decodeKey(schemaElement, value),
-      };
+    const schemaElement: ERC725JSONSchema = isDynamic
```

Comment:
Why this line was changed? Is it related to the bit count?
Feels like it might related to issue #372 or am I completely wrong?

Reply:
It just converts the expression using a tenary early so that it was easier to create a common exception handler.